server/controllers/message.js

```diff
@@ -0,0 +1,84 @@
+const asyncHandler = require("express-async-handler");
+const Conversation = require("../models/Conversation");
+const Message = require("../models/Message");
+const Profile = require("../models/Profile");
+
+// @route POST /messages
+// @desc send new message in existing conversation between authenticated user (sender), and other user (recipient)
+// @access private
+exports.createMessage = asyncHandler(async (req, res, next) => {
+  const { conversationId, text, recipientId } = req.body;
+  const userId = req.user.id;
+
+  const profile = await Profile.findOne({ user: userId });
+
+  const conversation = await Conversation.findOne({
+    _id: conversationId,
+    members: { $all: [profile._id, recipientId] },
+  });
+
+  if (!conversation) {
+    res.status(404).json({ error: "Conversation not found!" });
+  }
+
+  let message = await Message.create({
+    conversationId,
+    sender: profile._id,
+    text,
+  });
+
+  message = await Message.populate(message, {
+    path: "sender",
+    select: { firstName: 1, lastName: 1 },
+  });
+
+  if (message) {
+    res.status(201).json({
+      message,
+    });
+  } else {
+    res.status(500);
+    throw new Error("Internal Server Error");
+  }
+});
+
+// @route PATCH /messages/:id
+// @desc update message `seen` to true, when message is seen
+// @access private
+exports.updateMessageStatus = asyncHandler(async (req, res, next) => {
+  const { conversationId, recipientId } = req.body;
```

Comment:
Also, validate so you return a `400` if any of the fields is not supplied by the front-end